client/parser/scanner.go

```diff
@@ -664,7 +664,7 @@ func (s *scanner) scanToken() error {
 		// skip whitespace
 		return nil
 	case '#':
-		s.skipComment()
+		s.skipComment() // TODO: scan comments, only drop at the parsing level
```

Comment:
I thought a little bit about whether the AST should include comments. I think that it _has to_ include comments, in order to support code formatting, otherwise we would lose your comments when we reformat your code.

It seems like it might be useful if comments could be associated with other AST nodes, to potentially unlock annotation-like features in the future, so I'm thinking that maybe there could be a CommentedNode node, which combines an ASTNode with a comment string and a description of where, textually, the comment should go (is it on its own line? is it an inline comment at the end of the line?)

Reply:
Hmm so the question here is how do we determine the formatting for a comment (newline vs sameline)? 

One way is to have a commented node. If there is a commented node, it will go immediately after. Otherwise if it is a normal comment it's just on a newline. 

Another thing I was thinking about is dealing with custom formatting. Perhaps we should extend the scanner to capture extra whitespace. Then in our formatter there will be instances (such as around comments) where we leave the user-defined whitespace. I'm not sure this is necessary though, and I would be ok with just the concept of a commented node.